python/sidecar/typing.py

```diff
@@ -17,19 +18,21 @@ class Reference(RefStudioModel):
     """A reference for an academic paper / PDF"""
     source_filename: str
     filename_md5: str
+    citation_key: str | None = None
     title: str | None = None
     abstract: str | None = None
     contents: str | None = None
+    published_date: date | None = None
     authors: list["Author"] = []
     chunks: list["Chunk"] = []
     metadata: dict[str, Any] = {}
 
 
 class Author(RefStudioModel):
     full_name: str
-    given_name: str
-    surname: str
-    email: str
+    given_name: str | None = None
+    surname: str | None = None
+    email: str | None = None
```

Comment:
Grobid won't always have these fields, but from what I can tell `full_name` will always be present if it can parse any authors.

We'll be able to get a lot more consistent here once we start work on further augmenting our references with #140